src/hypercall.rs

```diff
@@ -102,28 +102,22 @@ pub fn open(mem: &MmapMemory, sysopen: &mut OpenParams, file_map: &mut UhyveFile
 	let requested_path_ptr = mem.host_address(sysopen.name).unwrap() as *const i8;
 	let mut flags = sysopen.flags & ALLOWED_OPEN_FLAGS;
 	if let Ok(guest_path) = unsafe { CStr::from_ptr(requested_path_ptr) }.to_str() {
+		// See: https://lwn.net/Articles/926782/
+		// See: https://github.com/hermit-os/kernel/commit/71bc629
```

Comment:
Why is this commit referenced here?

Reply:
We're not just implementing this restriction because it would not work on Linux exclusively and because Linux decided to make such a decision, we are also implementing it because "we" decided to explicitly forbid `O_CREAT|O_DIRECTORY` in Hermit as well. I think both links are important for context.

Reply:
Not quite sure if the second one is necessary. But we can stil merge it